broker/router panic when method other than POST/GET/PUT/DELETE is sent to gateway

Example curl using a method of 'POS' (like a typo of POST), but the method could be anything else here other than 'POST/GET/PUT/DELETE'

```
curl -v   -X 'POS'   -H 'Content-Type: application/json'   --data '{"jsonrpc": "2.0", "id": 1, "method": "initialize"}'   http://mcp.127-0-0-1.sslip.io:8001/mcp
```

Do this a few times, and the router/broker panics like this:

```
time=2025-11-27T11:21:16.868Z level=DEBUG msg="Sending MCP body routing instructions to Envoy: request_body:{response:{header_mutation:{set_headers:{header:{key:\"x-mcp-method\" raw_value:\"initialize\"}} set_headers:{header:{key:\"x-mcp-servername\" raw_value:\"mcpBroker\"}}} clear_route_cache:true}}"
time=2025-11-27T11:21:16.868Z level=INFO msg="terminate session id in jwt session manager" sesssion=""
time=2025-11-27T11:21:16.869Z level=DEBUG msg="Sending response header processing instructions to Envoy: response_headers:{response:{header_mutation:{}}}"
time=2025-11-27T11:21:16.873Z level=ERROR msg="[ext_proc] Process: Error receiving request" error="rpc error: code = Canceled desc = context canceled"
time=2025-11-27T11:21:24.210Z level=ERROR msg="[ext_proc] Process: Error receiving request" error="rpc error: code = Canceled desc = context canceled"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x940e20]

goroutine 398 [running]:
github.com/kagenti/mcp-gateway/internal/mcp-router.(*ExtProcServer).HandleResponseHeaders(0x400011a6c0, {0x1fe5e58, 0x40005e2cc0}, 0x40005da3c0, 0x0, 0x0)
	/workspace/internal/mcp-router/response_handlers.go:18 +0x180
github.com/kagenti/mcp-gateway/internal/mcp-router.(*ExtProcServer).Process(0x400011a6c0, {0x1fee818, 0x4000401150})
	/workspace/internal/mcp-router/server.go:126 +0x1a8
github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3._ExternalProcessor_Process_Handler({0x1ad4a40?, 0x400011a6c0}, {0x1feb068, 0x40005d8100})
	/go/pkg/mod/github.com/envoyproxy/go-control-plane/envoy@v1.36.0/service/ext_proc/v3/external_processor_grpc.pb.go:106 +0xdc
google.golang.org/grpc.(*Server).processStreamingRPC(0x400032a488, {0x1fe5e58, 0x40005e2c60}, 0x40005ce1a0, 0x4000566360, 0x3189b60, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.77.0/server.go:1714 +0xdcc
google.golang.org/grpc.(*Server).handleStream(0x400032a488, {0x1fe6a28, 0x40002c1a00}, 0x40005ce1a0)
	/go/pkg/mod/google.golang.org/grpc@v1.77.0/server.go:1836 +0xa7c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/go/pkg/mod/google.golang.org/grpc@v1.77.0/server.go:1063 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 411
	/go/pkg/mod/google.golang.org/grpc@v1.77.0/server.go:1074 +0x138
```

--- a/internal/mcp-router/server.go
+++ b/internal/mcp-router/server.go
@@ -48,6 +48,7 @@ func (s *ExtProcServer) OnConfigChange(_ context.Context, newConfig *config.MCPS
 func (s *ExtProcServer) Process(stream extProcV3.ExternalProcessor_ProcessServer) error {
 	var (
 		localRequestHeaders *extProcV3.HttpHeaders
+		requestID           string
 		streaming           = false
 		mcpRequest          *MCPRequest
 	)
@@ -62,9 +63,15 @@ func (s *ExtProcServer) Process(stream extProcV3.ExternalProcessor_ProcessServer
 		switch r := req.Request.(type) {
 		case *extProcV3.ProcessingRequest_RequestHeaders:
 			// TODO we are ignoring errors here
+			if r.RequestHeaders == nil {
+				return fmt.Errorf("no request headers present")
+			}
 			localRequestHeaders = r.RequestHeaders
 			responses, _ := s.HandleRequestHeaders(r.RequestHeaders)
-			s.Logger.Debug("[ext_proc ] Process: request headers", "local headers ", localRequestHeaders.Headers)
+			requestID = getSingleValueHeader(localRequestHeaders.Headers, "x-request-id")
+			path := getSingleValueHeader(localRequestHeaders.Headers, ":path")
+			method := getSingleValueHeader(localRequestHeaders.Headers, ":method")
+			s.Logger.Debug("[ext_proc ] Process: ProcessingRequest_RequestHeaders", "request id:", requestID, "path", path, "method", method)
 			for _, response := range responses {
 				s.Logger.Debug(fmt.Sprintf("Sending header processing instructions to Envoy: %+v", response))
 				if err := stream.Send(response); err != nil {
@@ -86,7 +93,7 @@ func (s *ExtProcServer) Process(stream extProcV3.ExternalProcessor_ProcessServer
 					}
 				}
 			}
-
+			s.Logger.Debug("[ext_proc ] Process: ProcessingRequest_RequestBody", "request id:", requestID)
 			if len(r.RequestBody.Body) > 0 {
 				if err := json.Unmarshal(r.RequestBody.Body, &mcpRequest); err != nil {
 					s.Logger.Error(fmt.Sprintf("Error unmarshalling request body: %v", err))
@@ -123,6 +130,10 @@ func (s *ExtProcServer) Process(stream extProcV3.ExternalProcessor_ProcessServer
 			continue
 
 		case *extProcV3.ProcessingRequest_ResponseHeaders:
+			if r.ResponseHeaders == nil || localRequestHeaders == nil {
+				return fmt.Errorf("no response headers or request headers")
+			}
+			s.Logger.Debug("[ext_proc ] Process: ProcessingRequest_ResponseHeaders", "request id:", requestID)
 			responses, _ := s.HandleResponseHeaders(stream.Context(), r.ResponseHeaders, localRequestHeaders, mcpRequest)
 			for _, response := range responses {
 				s.Logger.Debug(fmt.Sprintf("Sending response header processing instructions to Envoy: %+v", response))
@@ -138,7 +149,8 @@ func (s *ExtProcServer) Process(stream extProcV3.ExternalProcessor_ProcessServer
 			s.Logger.Error("[EXT-PROC] Unexpected response body processing request received",
 				"size", len(r.ResponseBody.GetBody()),
 				"end_of_stream", r.ResponseBody.GetEndOfStream(),
-				"note", "response_body_mode is set to NONE in EnvoyFilter - this should not occur")
+				"note", "response_body_mode is set to NONE in EnvoyFilter - this should not occur",
+				"request-id", requestID)
 			// Return empty response to satisfy the interface
 			response := &extProcV3.ProcessingResponse{
 				Response: &extProcV3.ProcessingResponse_ResponseBody{

